src/main/java/org/folio/innreach/domain/service/impl/RecordContributionServiceImpl.java

```diff
@@ -0,0 +1,189 @@
+    var eligibleItems = false;
+    var statisticalCodeIds = item.getStatisticalCodeIds();
+    var centralServer = getCentralServer(item);
+    if (centralServer == null) {
+      log.warn("Can't find central server for an item with id = {} and permanent location id = {}. Skipping evaluating this item.",
+        item.getId(), item.getPermanentLocationId());
+    } else {
+      var centralServerId = centralServer.getId();
+      var suppressionCode = validationService.getSuppressionStatus(centralServerId, statisticalCodeIds);
+      if (suppressionCode != null && !suppressionCode.equals('n')
+        || validationService.getItemCirculationStatus(centralServerId, item).equals(ContributionItemCirculationStatus.AVAILABLE)) {
+        eligibleItems = true;
+      }
+    }
+    return eligibleItems;
+  }
+
+  @Override
+  public void decontributeInventoryInstanceEvents(Instance instance) {
+    log.info("Decontributing items from instance with id = " + instance.getId());
+    instance.getItems().forEach(this::decontributeInventoryItemEvents);
+  }
+
+  @Override
+  public boolean evaluateInventoryHoldingForContribution(Holding holding) {
+    log.info("Evaluating inventory holding.");
+    var result = true;
+    var statisticalCodeIds = holding.getStatisticalCodeIds();
+    if (statisticalCodeIds.size() != 1) {
+      log.warn("Multiple statistical codes defined, instance with id = {} is to be decontributed.", holding.getId());
+      result = false;
+    } else {
+      var centralServer = getCentralServer(holding);
+      if (centralServer == null) {
+        log.warn("Can't find central server for an item with id = {} and permanent location id = {}.",
+          holding.getId(), holding.getPermanentLocationId());
+        result = false;
+      } else {
+
+        var centralServerId = centralServer.getId();
+        var suppressionCode = validationService.getSuppressionStatus(centralServerId, statisticalCodeIds);
+        if (suppressionCode != null && suppressionCode.equals('n')) {
+          log.warn("Holding with id = {} has \"Do not contribute\" statistical code, all associated items" +
+            " is to be decontributed from central server with id = {}.", holding.getId(), centralServerId);
+          result = false;
+        }
+      }
+    }
+    log.info("Inventory holding evaluation complete.");
+    return result;
+  }
+
+  @Override
+  public void decontributeInventoryHoldingEvents(Holding holding) {
+    log.info("Decontributing holding with id = " + holding.getId());
+    var centralServer = getCentralServer(holding);
+    Assert.isTrue(centralServer != null,
+      String.format("Can't find central server for a holding with id = %s and permanent location id = %s.", holding.getId(), holding.getPermanentLocationId()));
+    var centralCode = centralServer.getCentralServerCode();
+
+    innReachExternalService.deleteInnReachApi(centralCode, "/contribution/bib/" + holding.getHrid());
```

Comment:
holding itself is not being contributed, instead you should try to decontribute items linked to it